### PR TITLE
[wasm] Add beginnings of support for building with ICU.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by vargaz** [comment]
For some reason, the PackageReference tag only works when in a separate proj, not when its in mono.proj, so need a separate icu-init.proj.

**Review by kg — APPROVED** [review]

File: src/libraries/Native/Unix/System.Globalization.Native/pal_icushim.c [hunk 1]
```diff
@@ -18,6 +18,8 @@
 
 #include "pal_icushim.h"
 
+#ifndef DISABLE_SHARED
```
Comment (safern) [inline src/libraries/Native/Unix/System.Globalization.Native/pal_icushim.c:21]: Why use this ifdef. Could we instead use a more descripting if def like, TARGETS_BROWSER?

Comment (akoeplinger) [reply src/libraries/Native/Unix/System.Globalization.Native/pal_icushim.c:21]: Yeah I'm not sure what this is used for either, @vargaz was this just for testing?

Comment (akoeplinger) [reply src/libraries/Native/Unix/System.Globalization.Native/pal_icushim.c]: I tried locally without it and it worked fine, removed it for now so we can get this PR in.

Comment (safern) [reply src/libraries/Native/Unix/System.Globalization.Native/pal_icushim.c]: How can `Load_ICU` work? It uses `dlopen` is that even supported in wasm?

Comment (kg) [reply src/libraries/Native/Unix/System.Globalization.Native/pal_icushim.c]: wasm supports dlopen, though we're not using it

Comment (akoeplinger) [reply src/libraries/Native/Unix/System.Globalization.Native/pal_icushim.c]: it's not, but we're running in Invariant mode still. That will be addressed by Egor/Katelyn's PR.

Comment (safern) [reply src/libraries/Native/Unix/System.Globalization.Native/pal_icushim.c]: Ok, cool

**Review by safern — APPROVED** [review]

**Pull request merged**